### PR TITLE
Proposal to add element equals command

### PR DESCRIPTION
**Pull request by mgrybyk**
## Proposed changes

Check if two elements are equal (web only)
```
$('foo').equals($('foo')) // true
$('foo').equals($('bar')) // false
```

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by mgrybyk** [comment]
The use case might be element order change or making sure it has changed after previous action. 
I understand that it might be very specific use case and don't mind closing the PR.

1. I'll rename it
2. Will try to make it work on mobile

**Comment by mgrybyk** [comment]
A bit busy right now will come back to it as soon as I can. If anyone wants pick it up feel free!

File: packages/webdriverio/src/commands/element/isEqual.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    const browser = getBrowserObject(this)
+
+    // mobile native
+    if (browser.isMobile) {
+        const context = await browser.getContext()
+        if (context.toLowerCase().includes('native')) {
+            const currentId = this.elementId
+            // just call any command that would refetch element if needed
+            try {
+                await this.getTagName()
```
Comment (mgrybyk) [inline packages/webdriverio/src/commands/element/isEqual.js]: @christian-bromann do you think it's a correct way to do refetch?
I'm afraid that the element was found not with just `$(selector)` but with some custom strategy  or stuff like that

Comment (mgrybyk) [reply packages/webdriverio/src/commands/element/isEqual.js]: ignore it, I don't think we need to refetch anything at this point

File: packages/webdriverio/src/commands/element/isEqual.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+
```
Comment (christian-bromann) [inline packages/webdriverio/src/commands/element/isEqual.js]:
```suggestion
```

File: packages/webdriverio/src/commands/element/isEqual.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+/**
+ *
+ * Return true if the selected DOM-element matches with the provided one.
```
Comment (christian-bromann) [inline packages/webdriverio/src/commands/element/isEqual.js]: As this is also supported for mobile we shouldn't restrict the description to the DOM API.

**Pull request merged**